lighthouse-core/audits/seo/font-size.js

```diff
@@ -157,12 +157,19 @@ function findStyleRuleSource(baseURL, styleDeclaration, node) {
         source = `${url.href}`;
 
         if (range) {
-          // `stylesheet` can be either an external file (stylesheet.startLine will always be 0),
-          // or a <style> block (stylesheet.startLine will vary)
-          const absoluteStartLine = range.startLine + stylesheet.startLine + 1;
-          const absoluteStartColumn = range.startColumn + stylesheet.startColumn + 1;
-
-          source += `:${absoluteStartLine}:${absoluteStartColumn}`;
+          // Add the startLine/startColumn of the <style> element to the range, if stylesheet is inline.
+          // Just use the rule's location if a sourceURL magic comment is present (`hasSourceURL` is true).
+          const addHtmlLocationOffset = stylesheet.isInline && !stylesheet.hasSourceURL;
+
+          const line = addHtmlLocationOffset ?
+            range.startLine + stylesheet.startLine + 1 :
+            range.startLine + 1;
+          // The column the stylesheet begins on is only relevant if the rule is declared on the same line.
+          const column = addHtmlLocationOffset && range.startLine === 0 ?
+            range.startColumn + stylesheet.startColumn :
+            range.startColumn;
```

Comment:
just a style issue, but interleaving the offset makes this read a little more confusingly than it is. What about something like

```js
let line = range.startLine + 1;
let column = range.startColumn;

// Add the startLine/startColumn of the <style> element to the range, if stylesheet is inline
// and a sourceURL magic comment is not present (`hasSourceURL` is true).
const addHtmlLocationOffset = stylesheet.isInline && !stylesheet.hasSourceURL;
if (addHtmlLocationOffset) {
  line += stylesheet.startLine;
  column += stylesheet.startColumn;
}

source += `:${line}:${column}`;
```

Reply:
also may be worth calling out why `sourceURL` means you don't want the offset because it isn't obvious from this context, at least (admittedly I've paged all font-size stuff out of my brain :)

Reply:
done and done